src/game/overlay.c

```diff
@@ -281,10 +307,207 @@ static void Overlay_OnFPSTextRemoval(const TEXTSTRING *textstring)
     m_FPSText = NULL;
 }
 
+static float Overlay_Ease(int32_t cur_frame, int32_t max_frames)
+{
+    float ratio = cur_frame / (float)max_frames;
+    if (ratio < 0.5f) {
+        return 2.0f * ratio * ratio;
+    }
+    float new_ratio = ratio - 1.0f;
+    return 1.0f - 2.0f * new_ratio * new_ratio;
+}
+
+static void Overlay_DrawPickup3D(DISPLAY_PICKUP_INFO *pu)
```

Comment:
The equivalent function `Output_DrawUISprite` is in `output.c`. Should me move that function into overlay or move this into output?

Reply:
In my opinion both should actually stay where they are; Output_DrawUISprite exists as a proxy for S_Output_DrawSprite (which shouldn't be called anywhere outside output.c). The equivalent to this in the 3d space is Output_DrawPolygons which already resides in output.c.